### ISSUE TITLE
In Tutorial: include/require/use other fnl files

### ISSUE THREAD

**Issue by uvtc** [bakpakin/Fennel#89]
In the tutorial, it's not clear how to compile one's project if it were to consist of multiple .fnl files (for example, a ./main.fnl file, plus some ./lib/foo.fnl and ./lib/bar.fnl files).

I think it would be very useful if, in the Tutorial :: "Modules and multiple files" section there were a simple Fennel-only example of how to include/require/use a couple of other .fnl library/module files from your main my-proj.fnl file.

**Comment by FongHou** [comment]
I have a Makefile like this.

SRC=$(shell ls *.fnl lib/*.fnl)

all: $(SRC)

$(SRC):
        fennel --compile $@ > $(subst .,.,$(@D))/$(subst .fnl,,$(@F)).lua

.PHONY: $(SRC)

**Comment by uvtc** [comment]
> Can you be a bit more specific about what you're trying to accomplish

Yes. I want to create these two files:

~~~
main.fnl
    lib/
        foo.fnl
~~~

and in foo.fnl have a function. I don't know the proper way to do that (I don't see a `defn` in the tutorial), but I suspect it may be:

~~~lisp
(local do-stuff
       (lambda [x]
               (print "hi" (* x 2))))
~~~

What do I put into main.fnl to call that? Per your suggestion above I tried:

~~~lisp
(require :lib.foo)

(print "hi")
(lib.foo.do-stuff 3)
(print "bye")
~~~

and tried running it like so:

~~~
fennel --compile main.fnl > main.lua
lua main.lua
~~~

but it failed with "lua: main.lua:1: module 'lib.foo' not found:".

**Comment by uvtc** [comment]
@FongHou thanks for the makefile example!

**Comment by uvtc** [comment]
Yow! I didn't know you could run a fennel file directly like that!

    fennel main.fnl

It's not mentioned at https://fennel-lang.org/ nor in the README. I think that should be mentioned in both places! What is it doing? Ah... wait. That is probably what @technomancy was talking about when he wrote:

> It's your choice; you can choose to perform ahead-of-time compilation (like the makefile above) or run the Fennel compiler at require time. A recent change (in 9331bba) made it so that requiring Fennel modules works out of the box if you run your code with the fennel executable

Thanks for the fixes, Matt. What you describe above works for me.

I see that I can also substitute in `lambda` for `fn` in my lib/foo.fnl and it works as well. I'd assume that that's the better way to go, to get the arity checking.

Whatever the current best practice is for this, I think it belongs in the tutorial.

**Comment by thatismatt** [comment]
I agree re the documentation, also I wonder if `defn` should be removed from fennel.lua.

**Comment by technomancy** [comment]
Hah; it never occurred to me to say it explicitly that you could run Fennel files with the `fennel` executable, but now that you mention it, that's a great point. =)
